src/App.tsx

```diff
@@ -1,223 +1,275 @@
+  const [isLoadingTodos, setIsLoadingTodos] = useState(true);
+  const [isLoadingNewName, setIsLoadingNewName] = useState(true);
+
+  const areTodosCompleted = todos.find(todo => !todo.completed);
+
+  const updateTodoCompleted = (todoNew: Todo) => {
+    setTodos(
+      todos.map((todo) => (todo.id !== todoNew.id
+        ? todo
+        : { ...todo, completed: !todo.completed })),
+    );
+  };
+
+  const updateTODOCompleted = (
+    todoId: number | undefined,
+    completed: boolean,
+  ) => {
+    updateTodo(todoId, completed)
+      .then((todo) => updateTodoCompleted(todo))
+      .catch(() => setIsUpdate(true));
+  };
+
+  const updateTodoName = (todoNew: Todo) => {
+    setTodos(
+      todos.map((todo) => (todo.id !== todoNew.id
+        ? todo
+        : { ...todo, title: todoNew.title }
+      )),
+    );
+  };
+
+  const updateTODOTitle = (todoId: number | undefined, title: string) => {
+    updateTodoTitle(todoId, title)
+      .then((todo) => {
+        updateTodoName(todo);
+        setIsLoadingNewName(false);
+      })
+      .catch(() => setIsUpdate(true));
+  };
+
+  const completedTodo = todos.filter((todo) => todo.completed);
+
+  const clearCompleted = () => {
+    completedTodo.filter((todo) => deleteTodo(todo.id)
+      .then()
+      .catch(() => setIsDelete(true)));
+  };
+
+  const setStatusCompleted = () => {
+    todos.forEach(todo => updateTodo(todo.id, !todo.completed)
+      .then(() => setTodos(todos.map((tod) => ({ ...tod, completed: true }))))
+      .catch(() => setIsUpdate(true)));
+  };
+
+  const setStatusNotCompleted = () => {
+    todos.forEach(toDo => updateTodo(toDo.id, !toDo.completed)
+      .then(() => setTodos(todos.map((t) => ({ ...t, completed: false }))))
+      .catch(() => setIsUpdate(true)));
+  };
+
+  const updateTodos = (todoId: unknown) => setTodos(todos
+    .filter((todo) => todo.id !== todoId));
+
+  const removeTodo = (todoId: number | undefined) => {
+    deleteTodo(todoId)
+      .then((todoID) => updateTodos(todoID))
+      .catch(() => setIsDelete(true));
+  };
+
+  const addNewTodo = (todo: Todo) => setTodos([...todos, todo]);
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      if (value !== '') {
+        createTodo(value, user?.id)
+          .then((todo) => addNewTodo(todo))
+          .catch(() => setIsError(true));
+        setValue('');
+      } else {
+        setIsEmpty(true);
+      }
+    }
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    setIsHidden(true);
+    getTodos(user?.id)
+      .then((todoss) => {
+        setTodos(todoss);
+        setIsLoadingTodos(false);
+      })
+      .catch(() => setIsError(true));
+    setTimeout(() => {
+      setIsHidden(true);
+      setIsDelete(false);
+      setIsEmpty(false);
+    }, 3000);
+
+    setIsAdding(true);
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+  }, [todos]);
```

Comment:
```suggestion
  }, []);
```
try to remove it from dependencies